src/Workspaces/CoreTest/SolutionTests/SolutionWithSourceGeneratorTests.cs

```diff
@@ -592,5 +592,27 @@ public async Task FreezingSolutionEnsuresGeneratorsDoNotRun(bool forkBeforeFreez
 
             Assert.False(generatorRan);
         }
+
+        [Fact]
+        [WorkItem(56702, "https://github.com/dotnet/roslyn/issues/56702")]
+        public async Task ForkAfterFreezeWorks()
+        {
+            using var workspace = CreateWorkspaceWithPartialSemantics();
+            var analyzerReference = new TestGeneratorReference(new SingleFileTestGenerator("// Hello, Source Generators!"));
+            var project = AddEmptyProject(workspace.CurrentSolution)
+                .AddAnalyzerReference(analyzerReference)
+                .AddDocument("RegularDocument.cs", "// Source File", filePath: "RegularDocument.cs").Project;
+
+            // Ensure generators are ran
+            _ = await project.GetCompilationAsync();
+
+            var document = project.Documents.Single().WithFrozenPartialSemantics(CancellationToken.None);
```

Comment:
Why the generated document isn't in the project after generators are ran?

Reply:
project.Documents is only the non-generated ones, you'd have to call GetSourceGeneratedDocumentsAsync() to get those -- we can't update the list since no chance of async there.